plugins/obs-ffmpeg/obs-ffmpeg-mux.c

```diff
@@ -1144,14 +1141,13 @@ static void insert_packet(struct darray *array, struct encoder_packet *packet,
 		pkt.pts -= audio_dts_offsets[pkt.track_idx];
 	}
 
-	for (idx = packets.num; idx > 0; idx--) {
-		struct encoder_packet *p = packets.array + (idx - 1);
+	for (idx = packets->num; idx > 0; idx--) {
+		struct encoder_packet *p = packets->array + (idx - 1);
 		if (p->dts_usec < pkt.dts_usec)
 			break;
 	}
 
-	da_insert(packets, idx, &pkt);
-	*array = packets.da;
+	da_insert((*packets), idx, &pkt);
```

Comment:
I don't think the parentheses are needed in lines like this. It's just passed as is, so there's no added clarity, and instead I think it looks more complex. Especially since the third parameter used a `&` and also didn't use parentheses. There's only one similar circumstance in the repo https://github.com/obsproject/obs-studio/blob/a35e5aae4116f6570a72a3e30f57142e59566063/libobs/util/profiler.c#L468 (and two other calls in that same file) (you could search repo with `(\w+\(|,\s*)\(\*\w+\)[,)]`), but hundreds of occurrences without the parentheses in a function parameter (search with `(\w+\(|,\s*)\*\w+[,)]`). Such parentheses only makes sense imo for stuff like `(*p)->type` and `strtol((*argv)+3, ...)`

Reply:
> I don't think the parentheses are needed in lines like this.

You are right. Instead of having parentheses here, I think we need to have the parentheses inside the macro.

Reply:
oh, is that because the macro puts it in the sizeof call and uses its fields, so the parentheses are necessary? but yeah if it is okay to simply edit the macro to have the parentheses, i think that would be better for less confusion and for consistency with normal functions

Reply:
Feel free to add parentheses to the macros if that would make code cleaner

Reply:
@norihiro I would appreciate if you could write a separate PR to add the parentheses to the macro and have this PR depend on that (especially since this still a draft anyway), so https://github.com/obsproject/obs-studio/pull/9048 will not have the parentheses anymore

Reply:
nevermind, realized i should still add the parentheses in the docs, as plugins would build against older versions

Reply:
It's fine to have that change in this same PR.

Reply:
I don't see an approve changes on me, nor "Resolve conversation", so it seems that you are the one who needs to resolve this conversation, @norihiro, just in case you forgot about this